okhttp/src/commonMain/kotlin/okhttp3/internal/-HttpUrlCommon.kt

```diff
@@ -761,7 +772,13 @@ internal object CommonHttpUrl {
               }
             } else {
               host = input.percentDecode(pos = pos, limit = portColonOffset).toCanonicalHost()
-              port = HttpUrl.defaultPort(scheme!!)
+              val defaultPort: Int = HttpUrl.defaultPort(scheme!!)
```

Comment:
I'm not sure this is needed since we have already validated scheme is http or https.

Reply:
Sure, then please add a test that exercises these paths.